iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.cpp

```diff
@@ -80,9 +80,13 @@ std::shared_ptr<CachedQueryAdaptor> QueryAdaptorCache::TryGet(Utf8CP ecsql, bool
     ErrorListenerScope err_scope(const_cast<ECDb&>(m_conn.GetPrimaryDb()));
     if (usePrimaryConn)
         status = newCachedAdaptor->GetStatement().Prepare(m_conn.GetPrimaryDb(), ecsql, !suppressLogError);
-    else
-        status = newCachedAdaptor->GetStatement().Prepare(m_conn.GetPrimaryDb().Schemas(), m_conn.GetDb(), ecsql, !suppressLogError);
-
+    else {
+        if (m_doNotUsePrimaryConnToPrepare) {

```

Comment:
Just realized we already have the usePrimaryConn flag.
Putting a second m_doNotUsePrimaryConnToPrepare flag alongside seems really confusing. Especially since one is named positive and the other negated.
I suggest we call this m_allowPrepareOnSeparateThread.
I also wondered why one is given to this method while the other is pulled from config, but I guess one is global while the other can be set on a per-query basis.

Reply:
I agree that the names can be confusing. The `usePrimaryConn` flag indicates whether to run the query synchronously. The other flag controls how to prepare the query if it runs asynchronously, using either the primary schema cache or the thread connection cache. By default, this feature is off. During testing, I found an issue where, despite each thread having its own connection, there are moments when they serialize, causing longer result production times instead of being faster. I will log an issue, as it seems there's something causing the simultaneous use of `ECSqlStatement` to not be entirely concurrent. My guess is that it might be related to the functions we register, but this is beyond the scope of this PR.